src/Domain/DoctrineDonationRepository.php

```diff
@@ -185,6 +185,7 @@ public function findWithExpiredMatching(\DateTimeImmutable $now): array
             ->innerJoin('d.fundingWithdrawals', 'fw')
             ->where('d.donationStatus IN (:expireWithStatuses)')
             ->andWhere('d.createdAt < :expireBefore')
+            ->andWhere('d.mandate is null')
```

Comment:
I'm not sure I get why we need this condition vs. relying on the statuses?

Reply:
the 1st donation in a mandate would be `pending`, and for regular giving if the donor walks away from 3DS we don't want to just remove the match funds from it as this would do, we want to fully cancel it and cancel the associated mandate. I don't want a situation where a donor can lose out on matching without knowing it just by being slow to complete 3DS.

In practice with current numbers this wouldn't be an issue because the mandate donations are auto cancelled quicker than the ad-hoc donations have match funds released, but I don't want to rely on that here.

Reply:
OK, fair enough. Possibly worth a comment so it's clear why later?